index.js

```diff
@@ -119,22 +119,25 @@ function resolveLocalRelativeImport(fileImport, file) {
  * @param {String} rootPath: root path of the project code
  */
 function resolveWebpackShim(source, kibanaPath, rootPath) {
-  const pluginShimPath = path.join(rootPath, 'webpackShims');
-  const pluginMatches = getFileMatches(source, pluginShimPath);
-  const pluginFileMatches = getMatch(pluginMatches, pluginShimPath);
-  debug(`resolveWebpackShim: checking for ${source}`);
-  if (pluginFileMatches.found) {
-    debug(`resolved webpackShim import in plugin: ${source}`);
-    return pluginFileMatches;
-  }
+  debug(`resolveWebpackShim: resolving ${source}`);
+  const sourceParts = source.split('/');
+  const baseSource = sourceParts.pop();
+
+  const pluginShimPaths = [
+    path.join(rootPath, 'webpackShims', ...sourceParts),
```

Comment:
The `webpackShims` directory could be in any directory relative to the file up to the root, so these paths need to include checks for intermediate checks. Maybe something like this?

```js
const relativeWebpackShims = sourceParts.reduce((acc, sourcePart) => [
  ...acc,
  path.join(acc[acc.length - 1], sourcePart, 'webpackShims')
], [
  path.join(rootPath, 'webpackShims')
])

const pluginShimPaths = [
  ...relativeWebpackShims,
  path.join(kibanaPath, 'webpackShims', ...sourceParts)
]
```